cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -488,8 +511,10 @@ protected StatefulSet createStatefulSet(
                             .withAnnotations(getPrometheusAnnotations())
                         .endMetadata()
                         .withNewSpec()
+                            .withServiceAccountName(getServiceAccountName())
```

Comment:
Why do we need the service account now?

Reply:
The `KafkaCluster` needs this service account because the `kafka-init` container needs permissions for getting nodes information (i.e. value from the label specified as `topologyKey`). Maybe at one point we could have the need for permissions in other "init containers" for Zookeeper and Connect as well. For now they override the method just returning `null` so no service account.